tests/test_devshell.py

```diff
@@ -125,6 +125,11 @@ def test_handles_ignores_extra_fields(self):
       self.assertEqual('sometoken', creds.access_token)
 
   def test_refuses_to_save_to_well_known_file(self):
-    with _AuthReferenceServer():
-      creds = DevshellCredentials()
-      self.assertRaises(NotImplementedError, save_to_well_known_file, creds)
+    ORIGINAL_ISDIR = os.path.isdir
+    try:
+      os.path.isdir = lambda path: True
```

Comment:
Was there any particular reason for reassigning `os.path.isdir` as opposed to changing the cloud SDK environment variable (say, to a temporary directory)?